examples/TestDummy/filter-with-dummy.js

```diff
@@ -1,52 +1,45 @@
-function someDirective() {
-  var directive = {
-    restrict: 'E',
-    replace: true,
-    template: '<div> {{ text | capitalize }}</div>'
-  };
-  return directive;
-}
+'use strict';
 
 function capitalize() {
-  var directive = {
-    restrict: 'EA',
-    replace: true,
-    template: '<div></div>'
+  return function(text) {
+    return text.toUpperCase();
   };
-  return directive;
 }
 
 angular
 .module('AnotherModule3', [])
-.directive('capitalize', anotherElement);
+.filter('capitalize', capitalize);
+
+function someDirective() {
+  return {
+    restrict: 'E',
+    replace: true,
+    template: '<div>{{ text | capitalize }}</div>'
+  };
+}
 
 angular
 .module('filterWithDummy', ['AnotherModule3'])
 .directive('someDirective', someDirective);
 
 
 describe('someDirective', function() {
-  var
-    element, $compile, $rootScope, $scope;
-
-  beforeEach(module('filterWithDummy', {
-    capitalizeFilter: TestDummy.filter
-  }));
-
-  beforeEach(function() {
-    inject(function(_$compile_, _$rootScope_) {
-      $compile = _$compile_;
-      $rootScope = _$rootScope_;
-      $scope = $rootScope;
-      element = angular.element('<some-directive></some-directive>');
-      $compile(element)($scope);
-      angular.element(document.body).append(element);
-      $scope.$apply();
-    });
+  var Test, element, $scope;
+
+  module.sharedInjector();
+
+  beforeAll(function() {
+    // FIXME: Do we need the Dummy.Filter? It's easier to pass an empty function(){};
```

Comment:
Dummy filter and directive are only aliases. It's easier to remember words than choose between function and object.